usaspending_api/etl/subaward_etl.py

```diff
@@ -135,17 +147,25 @@ def load_subawards(submission_attributes, db_cursor):
                 d2_empty_count += 1
             continue
 
+        agency = get_valid_awarding_agency(row)
+
+        if not agency:
+            logger.warn("Subaward number {} cannot find matching agency with toptier code {} and subtier code {}".format(row['subaward_num'], row['awarding_agency_code'], row['awarding_sub_tier_agency_c']))
+            continue
+
         # Find the award to attach this sub-award to
         # We perform this lookup by finding the Award containing a transaction with
         # a matching fain and submission. If this fails, try submission and uri
         if row['fain'] and len(row['fain']) > 0:
-            award = Award.objects.filter(transaction__submission=submission_attributes,
+            award = Award.objects.filter(awarding_agency=agency,
```

Comment:
Mulling this over some more, I realize we use this same award lookup logic a few times in the code base. Perhaps not a thread we want to pull right now, but perhaps it would make sense to split `Award.get_or_create_summary_award` into two: `get_award` that does the lookup and can be used as needed and `create_award`.